social_dilemmas/envs/agent.py

```diff
@@ -148,99 +155,113 @@ def consume(self, char):
 
 
 HARVEST_ACTIONS = BASE_ACTIONS.copy()
-HARVEST_ACTIONS.update({7: 'FIRE'})  # Fire a penalty beam
-
-HARVEST_VIEW_SIZE = 7
+HARVEST_ACTIONS.update({7: "FIRE"})  # Fire a penalty beam
 
 
 class HarvestAgent(Agent):
-
-    def __init__(self, agent_id, start_pos, start_orientation, grid, view_len=HARVEST_VIEW_SIZE):
+    def __init__(self, agent_id, start_pos, start_orientation, full_map, view_len):
         self.view_len = view_len
-        super().__init__(agent_id, start_pos, start_orientation, grid, view_len, view_len)
+        super().__init__(agent_id, start_pos, start_orientation, full_map, view_len, view_len)
         self.update_agent_pos(start_pos)
         self.update_agent_rot(start_orientation)
 
-    @property
-    def action_space(self):
-        return Discrete(8)
-
     # Ugh, this is gross, this leads to the actions basically being
     # defined in two places
     def action_map(self, action_number):
         """Maps action_number to a desired action in the map"""
         return HARVEST_ACTIONS[action_number]
 
-    @property
-    def observation_space(self):
-        return Box(low=0.0, high=0.0, shape=(2 * self.view_len + 1,
-                                             2 * self.view_len + 1, 3), dtype=np.float32)
-
     def hit(self, char):
-        if char == 'F':
+        if char == b"F":
```

Comment:
what function does the b serve? It'd be nice to comment somewhere; might have missed it.

Reply:
In short:
```
Python 3.6.9 (default, Apr 18 2020, 01:56:04) 
[GCC 8.4.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import sys
>>> sys.getsizeof('F')
50
>>> sys.getsizeof(b'F')
34
```

The latter is a byte instead of a string (Python doesn't have chars). It's supposed to make the map memory footprint smaller. Conversion from the string map to the byte map happens in MapEnv.ascii_to_numpy. I'm not sure what the correct place for a comment would be - perhaps in that function, or in maps.py?

Reply:
Nah you know what, that makes sense!